doc/contributing/contributing.rst

```diff
@@ -31,49 +31,55 @@ To contribute to the source code of the model you will need to fork the reposito
 Quickstart Guide
 ----------------
 
-1. Fork the project on GitHub (using the fork button) and create a local clone, if you haven’t already:
+**1.** Fork the project on GitHub (using the fork button).
+
+**2.** Create a local clone (we strongly suggest keeping a separate repository for development work):
 
 ::
 
     % git clone https://github.com/user_name/MITgcm.git
 
-2. Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
+**3.** Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
 
 ::
 
     % git remote add upstream https://github.com/altMITgcm/MITgcm.git
 
-3. This step is optional, but highly recommended. It will ensure that your GitHub repo stays up to date with the main repository. Checkout the master branch and sync everywhere:
+**4.** Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a new feature) and make edits on this branch:
 
 ::
 
-   % git checkout master
-   % git pull upstream master
-   % git push origin master
-   
-4. Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a bugfix) and make edits on this branch:
+   % git fetch upstream
+   % git checkout -b newfeature upstream/master
 
-::
+**5.** When edits are done, do all git add’s and git commit’s. In the commit, make a succinct (<70 char) summary, followed by a blank line and a longer description. Reference any outstanding issues addressed using the syntax ``#ISSUE_NUMBER``.
```

Comment:
I don't think a longer description belongs in the commit history itself. The longer description can come when the user makes the actual PR via github. It is common for one PR to have many commits ([example](https://github.com/pydata/xarray/pull/1796/commits), which are then squashed at merge time. Each individual commit just needs the succinct summary.

Reply:
OK, how about we rephrase this to 
"In the commit message, make a succinct (<70 char) summary of your changes. If you need more space to describe your changes, you can leave a blank line and type a longer description, or break your commit into multiple smaller commits."